asaps/cli.py

```diff
@@ -15,56 +18,91 @@
               help='The password for authentication.')
 @click.option('-d', '--dry_run', prompt='Dry run?', default=True,
               help='Perform dry run that does not modify any records.')
-def main(url, username, password, dry_run):
+@click.pass_context
+def main(ctx, url, username, password, dry_run):
+    ctx.obj = {}
+    logger = structlog.get_logger()
```

Comment:
The call to `get_logger` can be done at the module level. This is a pretty standard practice. The main thing is that we want to do the actual configuring of the logger in a more controlled way as you've done here. By moving the `get_logger` call to the module level you avoid having to pass the logger around as an argument. Each module that uses logging would have a `get_logger` call at the top.